### PR TITLE
chore(core): improve the logging of assets feature

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
Seeing a lot of confusion about the `noop` logging. This issue has come up multiple times on Discord already (e.g., [Discord link](https://discord.com/channels/830184174198718474/1154082475547574385)). I believe we should adjust the logging to avoid confusing users.

## Changes

- **Changed** the logging approach in `validateAssetsFeature`
- **Removed** the error log stating that the `assets` feature is unsupported
- **Lowered** the severity of the log from error to warning for `sharp` or `squoosh` incompatibility. Since we automatically replace these with `noop`, the user shouldn't need to make any manual changes; therefore, an error is too severe.
- **Reworded** the warning log about automatically switching to `noop`

**For more details, see self-review comments**

## Docs

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

File: packages/astro/src/integrations/astroFeaturesValidation.ts
Comment (alexanderniebuhr) [inline packages/astro/src/integrations/astroFeaturesValidation.ts:134]: the user shouldn't need to make any manual changes; therefore, an error is too severe

File: packages/astro/src/integrations/astroFeaturesValidation.ts
Comment (alexanderniebuhr) [inline packages/astro/src/integrations/astroFeaturesValidation.ts:142]: the user shouldn't need to make any manual changes; therefore, an error is too severe

File: packages/astro/src/integrations/index.ts
Comment (alexanderniebuhr) [inline packages/astro/src/integrations/index.ts:228]: As already laid out in the comment, I believe that in the long term, this message should not be based on a boolean. Instead, we should be able to filter whether a feature is unsupported or partially supported.

File: packages/astro/src/integrations/index.ts
Comment (alexanderniebuhr) [inline packages/astro/src/integrations/index.ts:236]: the user shouldn't need to make any manual changes; therefore, an error is too severe

File: packages/astro/src/integrations/index.ts
Comment (alexanderniebuhr) [inline packages/astro/src/integrations/index.ts:238]: We don't need to mention `sharp` or `squoosh` again, as we've already stated in the log line above that the current adapter is incompatible with them.

**Review by alexanderniebuhr — COMMENTED** [review]
self-review for background context

File: packages/astro/src/integrations/index.ts
Comment (florian-lefebvre) [inline packages/astro/src/integrations/index.ts:225]:
```suggestion
								// in theory 'assets' false, doesn't mean that the feature is not supported, but rather that the chosen image service is unsupported
```

File: packages/astro/src/integrations/index.ts
Comment (Princesseuh) [reply packages/astro/src/integrations/index.ts]: Yeah, the support here is confusing. In my opinion, it would've been better for it to be more about the Node compatibility of the image service vs the specific services.

File: packages/astro/src/integrations/index.ts
Comment (sarah11918) [reply packages/astro/src/integrations/index.ts:238]: Just checking that people feel that "does not support image processing/disabling image processing" is clear enough that it won't make people think they somehow can't even use/have images in their project. (Out of context of the inner workings of the image services API, "processing" could literally mean "handling/being able to accept".)

It might be worth something slightly more reassuring that mentions what WILL happen to/with your images (which sounds less like images are somehow a problem): they will not be optimized, they will be copied into your build as is, this allows you to use the image component and `getImage()` function although the functionality will be limited to preventing CLS and will not optimize or otherwise process your images.

Comment (alexanderniebuhr) [reply packages/astro/src/integrations/index.ts:238]: @sarah11918 do you have a suggestion for me, or do we wait for other opinions?

Comment (ematipico) [reply packages/astro/src/integrations/index.ts:238]: We can say something like "The selected adapter ${adapter.name} doesn't support image optimization, although they will still be served correctly. To ensure your project ..."

**Review by ematipico — APPROVED** [review]
The PR looks good to me. Let's wait for Sarah's approval

File: packages/astro/src/integrations/index.ts
Comment (sarah11918) [inline packages/astro/src/integrations/index.ts:238]:
```suggestion
									`The selected adapter ${adapter.name} does not support image processing. To allow your project to build with the original, unprocessed images, the image service has been automatically switched to the 'noop' option.`
```
I'm wondering whether something like this might be sufficient (and more reassuring that you do get images)? 

If it's possible to link for more info somewhere, then this would be a good place https://docs.astro.build/en/reference/configuration-reference/#imageservice (and, I did make myself a docs issue to document a noop option!)

Comment (alexanderniebuhr) [reply packages/astro/src/integrations/index.ts]: I don't think we should add a link to the terminal log, but your version sounds better.

Comment (alexanderniebuhr) [reply packages/astro/src/integrations/index.ts]: I've updated it & also added the URL, should be working.

**Review by sarah11918 — APPROVED** [review]
I think we have a winner! I am sufficiently less frightened about my images as a reader, so docs is signing off!

**Pull request merged**
